systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaNodePoolST.java

```diff
@@ -0,0 +1,217 @@
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.junit.jupiter.api.Assumptions.assumeFalse;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Tag(REGRESSION)
+public class KafkaNodePoolST extends AbstractST {
+    private static final Logger LOGGER = LogManager.getLogger(KafkaNodePoolST.class);
+
+    /**
+     * @description This test case verifies transfer of Kafka Cluster from and to management by KafkaNodePool, by creating corresponding Kafka and KafkaNodePool custom resources
+     * and manipulating according kafka annotation.
+     *
+     * @steps
+     * 1. - Deploy Kafka with annotated to enable management by KafkaNodePool, and KafkaNodePool targeting given Kafka Cluster.
+     *    - Kafka is deployed, KafkaNodePool custom resource is targeting Kafka Cluster as expected.
+     * 2. - Modify KafkaNodePool by increasing number of Kafka Replicas.
+     *    - Number of Kafka Pods is increased to match specification from KafkaNodePool
+     * 3. - Produce and consume messages in given Kafka Cluster.
+     *    - Clients can produce and consume messages.
+     * 4. - Modify Kafka custom resource annotation strimzi.io/node-pool to disable management by KafkaNodePool.
+     *    - StrimziPodSet is modified, replacing former one, Pods are replaced and specification from KafkaNodePool (i.e., changed replica count) are ignored.
+     * 5. - Produce and consume messages in given Kafka Cluster.
+     *    - Clients can produce and consume messages.
+     * 6. - Modify Kafka custom resource annotation strimzi.io/node-pool to enable management by KafkaNodePool.
+     *    - new StrimziPodSet is created, replacing former one, Pods are replaced and specification from KafkaNodePool (i.e., changed replica count) has priority over Kafka specification.
+     * 7. - Produce and consume messages in given Kafka Cluster.
+     *    - Clients can produce and consume messages.
+     *
+     * @usecase
+     * - kafka-node-pool
+     */
+    @ParallelNamespaceTest
+    void testKafkaManagementTransferToAndFromKafkaNodePool(ExtensionContext extensionContext) {
+
+        final TestStorage testStorage = new TestStorage(extensionContext);
+
+        final int originalKafkaReplicaCount = 3;
+        final int nodePoolIncreasedKafkaReplicaCount = 4;
+        final String kafkaNodePoolName = "kafka";
+
+        // setup clients
+        KafkaClients clients = new KafkaClientsBuilder()
+            .withProducerName(testStorage.getProducerName())
+            .withConsumerName(testStorage.getConsumerName())
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(testStorage.getClusterName()))
+            .withTopicName(testStorage.getTopicName())
+            .withMessageCount(testStorage.getMessageCount())
+            .withDelayMs(200)
```

Comment:
why 200ms?

Reply:
as there are 3 checks like these I did not consider it necessary to drag for longer mainly as there are also additional stability checks.